File: Assets/Packages/Scripts/imported scripts/PaddleAutoController.lua
function PaddleAutoController:update(dt)
	if self:isPaddleAboveBall() then
		self.paddle.dy = self.paddleSpeed
	elseif self:isPaddleBelowBall() then
		self.paddle.dy = -self.paddleSpeed
	else
		self.paddle.dy = 0
	end
end

function PaddleAutoController:isPaddleAboveBall()
	return self.paddle.y + self.paddle.height < self.ball.y
end

function PaddleAutoController:isPaddleBelowBall()
	return self.paddle.y > self.ball.y + self.ball.height
end

File: Assets/Packages/GameManager/Scripts/PaddleAutoController.cs
using UnityEngine;

public class PaddleAutoController
{
    [SerializeField] Ball ball;
    [SerializeField] Paddle paddle;
    [SerializeField] float basePaddleSpeed;
    [SerializeField] float accelerationFactor = 0.9f;
    [SerializeField] float decelerationFactor = 0.1f;
    public PaddleAutoController(Ball ball, Paddle paddle, float basePaddleSpeed, float accelerationFactor = 0.9f, float decelerationFactor = 0.1f)
    {
        this.ball = ball;
        this.paddle = paddle;
        this.basePaddleSpeed = basePaddleSpeed;
        this.accelerationFactor = accelerationFactor;
        this.decelerationFactor = decelerationFactor;
    }
    public void Update(float dt)
    {
        float newSpeed = paddle.CurrentSpeed;
        if (IsPaddleAfterBall())
            newSpeed += basePaddleSpeed * accelerationFactor * dt;
        else if (IsPaddleBeforeBall())
            newSpeed -= basePaddleSpeed * accelerationFactor * dt;
        else
        {
            if (paddle.CurrentSpeed > 0)
                newSpeed -= basePaddleSpeed * decelerationFactor * dt;
            else if (paddle.CurrentSpeed < 0)
                newSpeed += basePaddleSpeed * decelerationFactor * dt;
        }
        paddle.SetCurrentSpeed(newSpeed);
    }
    public bool IsPaddleAfterBall() => paddle.GetAxisPosition() > ball.PositionParallelToPlayers;
    public bool IsPaddleBeforeBall() => paddle.GetAxisPosition() < ball.PositionParallelToPlayers;
}